mllib/src/main/scala/org/apache/spark/ml/fpm/FPGrowth.scala

```diff
@@ -217,7 +222,9 @@ object FPGrowth extends DefaultParamsReadable[FPGrowth] {
 @Experimental
 class FPGrowthModel private[ml] (
     @Since("2.2.0") override val uid: String,
-    @Since("2.2.0") @transient val freqItemsets: DataFrame)
+    @Since("2.2.0") @transient val freqItemsets: DataFrame,
+    private val itemSupport: scala.collection.Map[Any, Double],
```

Comment:
I suppose there's no way of adding the item generic type here; it's really in the schema of the DataFrame. Does `Map[_, Double]` also work? I don't think it needs a change, just a side question.

If you have the support for every item, do you need the overall count here as well? the item counts have already been divided through by numTrainingRecords here. Below only itemSupport is really passed somewhere else.

Reply:
Yes, `Map[_, Double]` seems to work too. I can change it if you prefer.

The `numTrainingRecords` is used because it needs to be stored (and it then used when loading the model in order to recompute the `itemSupport`).